### PR TITLE
[ONNX][dynamo_export] Skip instance_norm decomp for export

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120866

Otherwise, instance_norm is decomposed into batch_norm with training set to True.
Downstream exporter has no way to figure out that training is actually not needed.
On the other hand, ONNX does have InstanceNormalization operator defined, however
due to decomp, it unnecessarily exports as batch norm and glue code.

Depends on https://github.com/microsoft/onnxscript/pull/1284

### CODE REVIEW

File: test/onnx/test_fx_op_consistency.py [hunk 1]
```diff
@@ -1548,6 +1538,11 @@ def skip_torchlib_forward_compatibility(
             "Reshape", "empty tensor"
         ),
     ),
+    xfail(
+        "nn.functional.instance_norm",
+        model_type=pytorch_test_common.TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAM,
```
Comment (BowenBao) [inline test/onnx/test_fx_op_consistency.py:1543]: cc @titaiwangms, similar to what @liqunfu observed the other day, I need to move this xfail downwards to here to correctly only xfail on 'TORCH_EXPORT_EXPORTEDPROGRAM'.

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by thiagocrepaldi** [comment]
assuming that manipulating the decomposition table is not enough, we should discuss with Meta a mechanism to prevent decomposition to happen?

File: test/onnx/test_fx_op_consistency.py (hunk 1 above)
Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:1543]: Tracked by https://github.com/pytorch/pytorch/issues/120918.
I will use this to see the case.

File: test/onnx/test_fx_op_consistency.py [hunk 2]
```diff
@@ -1548,6 +1538,11 @@ def skip_torchlib_forward_compatibility(
...
+        reason="fixme: Assertion error: result mismatch",
```
Comment (titaiwangms) [inline test/onnx/test_fx_op_consistency.py:1544]: Is it because decomp slipping does not work with `ExportedProgram`, it fails?

**Review by titaiwangms — APPROVED** [review]
Such a quick support!

File: test/onnx/test_fx_op_consistency.py (hunk 2 above)
Comment (BowenBao) [reply test/onnx/test_fx_op_consistency.py:1544]: correct

Comment (BowenBao) [reply test/onnx/test_fx_op_consistency.py:1544]: Well, to be more precise, decomp skipping were not enabled during `torch.export`. It is only enabled during `torch.onnx.dynamo_export`.

File: test/onnx/test_fx_op_consistency.py (hunk 1 above)
Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:1543]: Revisiting on this. I think with matcher seems reasonable to put it down here though.

Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:1543]: I just confirmed that model_type only works with subfails.
